src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -111,6 +113,57 @@ public void setMaxWebSocketBufferSize(int maxWebSocketBufferSize) {
 		}
 	}
 
+	/**
+	 * Returns the handler that will be run when the listener disconnects unexpectedly due to potential error.
+	 * The listener will attempt to reconnect after this handler runs.	
+	 */
+	public Consumer<Throwable> getConnectingHandler() {
+		return connectingHandler;
+	}
```

Comment:
Since this is touching a bunch of properties would you consider putting the set/get "properties" after the constructors?

Reply:
I should probably make another PR just to move all the constructors, since there are multiple classes with constructors need to be relocated.

---
In reply to: [261314508](https://github.com/Azure/azure-relay-java/pull/26#discussion_r261314508) [](ancestors = 261314508)